examples/diff.c

```diff
@@ -290,9 +290,31 @@ static void parse_opts(struct opts *o, int argc, char *argv[])
 			o->findopts.flags |= GIT_DIFF_FIND_COPIES;
 		else if (!strcmp(a, "--find-copies-harder"))
 			o->findopts.flags |= GIT_DIFF_FIND_COPIES_FROM_UNMODIFIED;
-		else if (is_prefixed(a, "-B") || is_prefixed(a, "--break-rewrites"))
+		else if (is_prefixed(a, "-B") || is_prefixed(a, "--break-rewrites")){
 			/* TODO: parse thresholds */
 			o->findopts.flags |= GIT_DIFF_FIND_REWRITES;
+			if(a[2]=='/'){
+				uint16_t m = 0;
+				int i = 3;
+				while(a[i]!='%' || a[i]!=' '){
+					m *= 10;
+					m += a[i]-'0';
+					i++;
+				}
+				if(m<=100) o->findopts.break_rewrite_threshold = m;
+				//else there is error, show message;
+			}else if(a[2]!=' '){
```

Comment:
On Unix, the shell splits your arguments for you.  The only time you would have a space in `a` is if somebody literally quoted it.  For example:  `git diff "-B 20%"`.  Which nobody would likely do.  Instead, if you ran:  `git diff -B 20%`, then `20%` is the _next argument_ and this current `a` would not have a space in it.  It would simply end.  That's why using the helper functions that already exist to understand this sort of thing is very helpful.

This is illustrated very quickly by building this and running it on a POSIX-type environment - running `-B 20%` crashes immediately.  I would recommend you download some Linux distribution and run it in a VM, as we want to make sure that libgit2 and the examples work everywhere.